### PR TITLE
feat: notification filters

### LINKED ISSUE #264: Allow for simple reason filtering
Using Github API's reason field
https://developer.github.com/v3/activity/notifications/
screen shot 2017-02-15 at 4 43 02 pm

Do a quick filtering to only show certain type of notification. 
I am thinking a simple toggle of yes or no to included each type of reason.

ie. I can turn off all the notification except for the one specially @mentioned me

### PR DESCRIPTION
**Pull request by setchy**
New `Filter` sidebar route which contains any post filtering controls, including

- show bot notifications - moved from Settings
- reason filtering - new. closes #264 

Screenshot 2024-06-21 at 10 39 42 AM

When any non-default filters are selected, it will show a visual indicator
Screenshot 2024-06-21 at 10 40 24 AM

### CODE REVIEW

File: src/routes/Filters.test.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+          >
+            <MemoryRouter>
+              <FiltersRoute />
+            </MemoryRouter>
+          </AppContext.Provider>,
+        );
+      });
+
+      fireEvent.click(screen.getByLabelText('Go Back'));
+      // TODO - add expects fetch notifications to be called.  pending #1305
```
Comment (setchy) [inline src/routes/Filters.test.tsx:54]: Pending: #1305 to do a few final changes in this PR

**Comment by bmulholland** [comment]
I guess the settings store doesn't support storing arrays?

**Comment by setchy** [comment]
> I guess the settings store doesn't support storing arrays?

it can with a bit of work 😅

**Comment by setchy** [comment]
> I guess the settings store doesn't support storing arrays?

updated to use an array `Reason[]` instead of a nasty string 🙈

**Review by bmulholland — APPROVED** [review]
Thanks! I think the extra work to make it an array was worth it :)

**Comment by setchy** [comment]
> Thanks! I think the extra work to make it an array was worth it :)

100000%

**Comment by setchy** [comment]
@afonsojramos - looking forward to your feedback on this, too, before I merge

**Comment by setchy** [comment]
small update... changed the `reset to default filters` to match other footer styling

Screenshot 2024-06-28 at 8 18 55 AM

**Comment by bmulholland** [comment]
FWIW, in that last screenshot, there's no explanation text... I was looking for it and it still look me a minute to find the icon. Without the context, I'd never have guessed that's what the button does!

**Comment by afonsojramos** [comment]
What about adding a counter to the filter icon on the sidebar that when hovered showed a tooltip with the filters being used which could be individually removed AND cleared?

File: src/routes/Filters.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    <div className="flex h-screen flex-col" data-testid="filters">
+      <Header fetchOnBack={true}>Filters</Header>
+      <div className="flex-grow overflow-x-auto px-8">
+        <fieldset className="mb-3">
+          <legend id="notifications" className="mb-1 mt-2 font-semibold">
+            Users
+          </legend>
+          <Checkbox
+            name="showBots"
+            label="Show notifications from Bot accounts"
```
Comment (afonsojramos) [inline src/routes/Filters.tsx:45]: What if we change this into a "Hide notifications from Bot Accounts" instead? This way all filters would be disabled by default.

Comment (setchy) [reply src/routes/Filters.tsx:45]: Good suggestion. I'll make that change shortly

Comment (setchy) [reply src/routes/Filters.tsx]: Renamed the setting and updated labels as suggested

**Review by afonsojramos — APPROVED** [review]
Looks awesome!! 🚀 Excited for the next release 👀

**Pull request merged**